cartography/intel/gcp/iam.py

```diff
@@ -638,64 +638,163 @@ def load_bindings(neo4j_session: neo4j.Session, bindings: List[Dict], project_id
+                        project_id, gcp_update_tag,
+                    )
+
+                elif member.startswith('serviceAccount:'):
+                    is_deleted = True
+                    attach_role_to_service_account(
+                        neo4j_session,
+                        role_id, f"projects/{project_id}/serviceAccounts/{member[len('serviceAccount:'):]}",
+                        project_id,
+                        gcp_update_tag,
+                        is_deleted,
+                    )
+
+                elif member.startswith('group:'):
+                    grp = member[len('group:'):]
+                    group = {
+                        "id": f'projects/{project_id}/groups/{grp}',
+                        "email": grp,
+                        "name": grp.split('@')[0],
+                        "is_deleted": True
+                    }
+                    attach_role_to_group(
+                        neo4j_session, role_id,
+                        group,
+                        project_id, gcp_update_tag,
+                    )
+
+                elif member.startswith('domain:'):
+                    dmn = member[len('domain:'):]
+                    domain = {
+                        "id": f'projects/{project_id}/domains/{dmn}',
+                        "email": dmn,
+                        "name": dmn,
+                        "is_deleted": True
+                    }
+                    attach_role_to_domain(
+                        neo4j_session, role_id,
+                        domain,
+                        project_id,
+                        gcp_update_tag,
+                    )
+
 
 @timeit
 def attach_role_to_user(
-    neo4j_session: neo4j.Session, role_id: str, user_id: str,
+    neo4j_session: neo4j.Session, role_id: str, user: Dict,
     project_id: str, gcp_update_tag: int,
 ) -> None:
     ingest_script = """
+    MERGE (user:GCPUser{id:{UserId}})
+    ON CREATE SET
+    user:GCPPrincipal,
+    user.firstseen = timestamp()
+    SET
+    user.email = {UserEmail},
+    user.name = {UserName},
+    user.lastupdated = {gcp_update_tag},
+    user.isDeleted = {isDeleted}
+    WITH user
     MATCH (role:GCPRole{id:{RoleId}})
-    MATCH (user:GCPPrincipal{id:{UserId}})
     MERGE (user)-[r:ASSUME_ROLE]->(role)
     ON CREATE SET r.firstseen = timestamp()
     SET r.lastupdated = {gcp_update_tag}
+    WITH user
+    MATCH (p:GCPProject{id: {project_id}})
+    MERGE (p)-[pr:RESOURCE]->(user)
+    ON CREATE SET
+    pr.firstseen = timestamp()
+    SET pr.lastupdated = {gcp_update_tag}
     """
 
     neo4j_session.run(
         ingest_script,
         RoleId=role_id,
-        UserId=user_id,
+        UserId=user['id'],
+        UserEmail=user['email'],
+        UserName=user['name'],
+        isDeleted=user.get('is_deleted', False),
         gcp_update_tag=gcp_update_tag,
+        project_id=project_id,
     )
 
 
 @timeit
 def attach_role_to_service_account(
```

Comment:
do we need this new parameter `is_deleted`? we should follow the same pattern as attach_role_to_user, where we read the `is_deleted` from the user object.  In this case, we will read from service account.